rcl/src/rcl/timer.c

```diff
@@ -39,7 +39,7 @@ typedef struct rcl_timer_impl_t
   // The user supplied callback.
   atomic_uintptr_t callback;
   // This is a duration in nanoseconds.
-  atomic_uint_least64_t period;
+  atomic_int_least64_t period;
```

Comment:
I think you are addressing https://github.com/ros2/rcl/issues/967 on this PR too, but probably it would be nice to open different PR for this. and i think https://github.com/ros2/rcl/issues/967#issuecomment-1067964114 makes more sense to me.